### DISCUSSION TITLE
Regarding ikelifetime and sa lifetime.

### DISCUSSION THREAD

**Discussion by gowdamanil** [strongswan/strongswan#2206] [Q&A]
Hello tobias,

I wanted to ask how to set ikelifetime and sa lifetime for swanctl.conf, I configured for ipsec.conf
conn %default
    **ikelifetime=8h
    lifetime=12h**
    keyingtries=%forever
    
    
   Can u suggest how to do the bold marked ones in swanctl.conf, I am unable to do so.I saw the documentation pages, but got confused,can u help?
   
   
Thanks.

Another is what would be ideal timeout for dpd_delay and dpd_timeout for host to host transport mode connection.

**Comment by tobiasbrunner** [comment]
> I am unable to do so.I saw the documentation pages, but got confused,can u help?

What confuses you? Also, [this page](https://wiki.strongswan.org/projects/strongswan/wiki/fromipsecconf) on the old wiki might help.

> Another is what would be ideal timeout for dpd_delay and dpd_timeout for host to host transport mode `connection.

DPD settings really depend on your preferences, but note that `dpd_timeout` has no effect on IKEv2 connections, have a look at the [retransmission settings](https://docs.strongswan.org/docs/5.9/config/retransmission.html) instead.

**Reply by gowdamanil** [reply]
> DPD settings really depend on your preferences, but note that dpd_timeout has no effect on IKEv2 connections, have a look at the [retransmission settings](https://docs.strongswan.org/docs/5.9/config/retransmission.html) instead.

I am using ikev1 not ikev2

**Reply by gowdamanil** [reply]
>What confuses you? Also, [this page](https://wiki.strongswan.org/projects/strongswan/wiki/fromipsecconf) on the old wiki might help.

Thanks, got the above one, here the equivalent, conn.<conn>.rekey_time can be used to do the same for ikelifetime.

**Reply by tobiasbrunner** [reply]
> I am using ikev1 not ikev2

You shouldn't.

> here the equivalent, conn..rekey_time can be used to do the same for ikelifetime

It's not exactly the same.
